tests/builder.test.cpp

```diff
@@ -1186,97 +1229,52 @@ TEST(Builder, AddIngredientToBuilderUsingBasePathWithManifestContainingPlacedAct
     std::string result;
     result = c2pa::read_ingredient_file(ingredient_source_path, temp_dir);
 
-    // Extract the instance_id from the ingredient JSON, we need it to link the ingredient
-    std::string instance_id;
-    size_t instance_id_start = result.find("\"instance_id\":");
-    if (instance_id_start != std::string::npos) {
-        instance_id_start = result.find("\"", instance_id_start + 14); // Skip "instance_id":
-        if (instance_id_start != std::string::npos) {
-            instance_id_start++; // Skip the opening quote
-            size_t instance_id_end = result.find("\"", instance_id_start);
-            if (instance_id_end != std::string::npos) {
-                instance_id = result.substr(instance_id_start, instance_id_end - instance_id_start);
-            }
-        }
-    }
+    // Parse ingredient JSON and extract the instance_id
+    json ingredient_json = json::parse(result);
+    std::string instance_id = ingredient_json["instance_id"];
 
-    // Initialize the near-complete manifest JSON (contains ingredient JSON and placed action)
-    // We are going to replace the placeholder value in the "ingredientIds" array with 
-    // the ingredientId we got by reading the ingredient from file.
-    std::string manifest = R"({
-        "vendor": "a-vendor",
-        "claim_generator_info": [
+    // Create the manifest JSON structure with the placed action
+    json manifest_json = {
+        {"vendor", "a-vendor"},
+        {"claim_generator_info", json::array({
             {
-                "name": "c2pa-c test",
-                "version": "1.0.0"
+                {"name", "c2pa-c test"},
+                {"version", "1.0.0"}
             }
-        ],
-        "assertions": [
+        })},
+        {"assertions", json::array({
```

Comment:
Yes this is what I was referring to above, it would be great if we can define the ingredient like this, it's much more readable.